enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+
+TBD
+
+## Design Details
+
+### Open Questions
+
+### Load balancing and IO impact
+- What would be the IO impact or additional load caused by a recurring execution of the backup pod and taking the snapshot?
+  - If significant, should we avoid selecting the leader to reduce the risk of overloading the leader and triggering a leader election?
+  - We can find the leader via the [endpoint status](https://etcd.io/docs/v3.5/dev-guide/api_reference_v3/#message-statusresponse-apietcdserverpbrpcproto) but the leader can change during the course of the backup or we may not have the option of filtering the leader out if the backup pod is unschedulable on non-leader nodes (e.g low disk space). 
+- Other than the leader IO considerations, when picking where to run and save the backup, do we need to load balance or round-robin across all control-plane nodes?
+  - Leaving it to the scheduler to schedule the backup pod (or having the operator save on the node it's running on) is simpler, but we could end up with all/most backups being saved on a single node. In a disaster recovery scenario we could lose that node and the backups saved.
+  - Could be a future improvement if we start with the simpler choice of leaving it to the scheduler.
+
+### Test Plan
+
+Prior to merging the backup controller implementation, an e2e test would be added to openshift/origin to test the validity of the backups generated from the automated backups feature. The pass rate on this test will feature gate the API until we pass an acceptable threshold to make it available by default.
+
+This test would effectively run through the backup and restore procedure as follows:
+- Start with a cluster that has the `EtcdBackups` API enabled
+- Save a backup of the cluster by triggering a one time backup using the `EtcdBackups` API, or create a schedule to do that
+- Modify the cluster state post backup (e.g create pods, endpoints etc) so we can later ensure that restoring from backup does not include this additional state
+- Induce a disaster recovery scenario of 2/3 nodes lost with etcd quorum loss
+- Step through the recovery procedure from the saved backup
+- Ensure that the control-plane has recovered and is stable
+- Validate that we have restored the cluster to the previous state so that it does not have the post-backup changes
+
+See the [restore test design doc](https://docs.google.com/document/d/1NkdOwo53mkNBCktV5tkUnbM4vi7bG4fO5rwMR0wGSw8/edit?usp=sharing) for a more detailed breakdown on the restore test and the validation requirements.
+
+Along with the e2e restore test, comprehensive unit testing of the backup controller will also be added to ensure the correct reconciliation and updates of the:
+- Backup schedule
+- Backup retention
+- Backup status
+- Alerting rules
+
+### Graduation Criteria
+
+**TODO:** Need to clarify how feature gating relates to dev preview -> tech preview -> GA.
+
+#### Dev Preview -> Tech Preview
+
+- TBD
+
+#### Tech Preview -> GA
+
+- TBD
```

Comment:
this is known. Track record of reliable restore.  I propose 99% on all platforms as a starting point since 99% of our tests currently pass at this threshold and consequences of failed restore is extremely high.